arch/Kconfig

```diff
@@ -462,6 +462,18 @@ config FP_SHARING
 
 endif # FLOAT
 
+config MPU_ROM_REGION_SIZE
+	int "Size of read-only/executable region in memory"
+	depends on !XIP
+	depends on MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT
+	depends on MEMORY_PROTECTION
```

Comment:
Is there a sensible default that could be used here? `int` symbols default to an empty value instead of 0, which causes warnings when the `.config` is read back in.

Reply:
Is there a way to make this a required configuration value?
In other words, if the depends criteria are met, and no value supplied, throw an error, rather than fill in 0 and generate a warning?

Reply:
No way to have required values like that. Would get tricky to define. There's a bunch of utilities like `allyesconfig`, and it'd have to make sense for those too.

It's kinda-sorta required now though, in that you get a warning-turned-error if you load a configuration file that was saved without specifying it. Might not be the nicest kind of required though, since it also prevents the `menuconfig` from being run (see https://github.com/zephyrproject-rtos/zephyr/issues/9573, IMO that one's mostly a CMake issue).

Reply:
Why only having the option for a MPU_ROM_REGION for power of two MPU?
If the issue is to solve alignment of the 'rom region' in case of power-of-two, isn't there another way or a more clear name instead of MPU_ROM_REGION_SIZE?

Reply:
> Why only having the option for a MPU_ROM_REGION for power of two MPU?

@ruuddw because if the MPU does not require power of two region alignment, there is no problem and none of this is necessary.
It's not clear to me what your objections are.